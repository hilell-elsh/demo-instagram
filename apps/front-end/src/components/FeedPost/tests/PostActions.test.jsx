import React from 'react'
import ReactDOM from 'react-dom'
import {act , isElementOfType} from 'react-dom/test-utils'


describe("PostActions Component", ()=> {
    let PostActions, container
    beforeEach(async ()=> {
        vi.mock('../PostActions/ActionLike.jsx')
        vi.mock('../PostActions/ActionAddComment.jsx')
        vi.mock('../PostActions/ActionShare.jsx')
        vi.mock('../PostActions/ActionEpandMore.jsx')

        container = document.createElement('div')

        document.body.innerHTML= ''
        document.body.appendChild(container)

        PostActions = (await import('../PostActions.jsx')).default
    }) 

    test('should exist' , () => {
        expect(PostActions).toBeTruthy()
        expect(typeof(PostActions)).toBe('function')
    })

    test('should be rendered', () => {
        act(() => {
            ReactDOM.render(<PostActions />, container)
        })
        expect(container).toMatchSnapshot()
    })

    test('should be expended', () => {
        act(() => {
            ReactDOM.render(<PostActions expanded={true} />, container)
        })
        expect(container.querySelector('.action-expand-more-mock').getAttribute('data-expanded')).toBe('true')
    })

    test('should not be expended', () => {
        act(() => {
            ReactDOM.render(<PostActions expanded={false} />, container)
        })
        expect(container.querySelector('.action-expand-more-mock').getAttribute('data-expanded')).toBe('false');
    })

    test('should call setExpanded when emitted from ActionExpandMore', () => {
        const expandedCallback = vi.fn()
        
        act(() => {
            ReactDOM.render(<PostActions expanded={false} setExpanded ={expandedCallback}/>, container)
        })
       expect(expandedCallback).not.toBeCalled()

       container.querySelector('.action-expand-more-mock').click()
       expect(expandedCallback).toBeCalledTimes(1)
    })

})