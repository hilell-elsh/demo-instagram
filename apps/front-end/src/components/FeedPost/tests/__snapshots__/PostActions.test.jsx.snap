// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostActions Component should be rendered 1`] = `
<div>
  <div
    class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
  >
    <div />
    <div
      class="action-share-mock"
    />
    <div />
    <div
      class="action-expand-more-mock"
      data-expanded="undefined"
    />
  </div>
</div>
`;
